projects/igniteui-angular/src/lib/grid/grid.component.ts

```diff
@@ -399,6 +402,24 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
     @Input()
     public columnHidingTitle = '';
 
+    @Input()
+    get columnPinning() {
+        return this._columnPinning;
+    }
+
+    set columnPinning(value) {
+        this._columnPinning = value;
```

Comment:
Please add a check whether the new value is different than the current value and apply the new value only if there is need.